base/path.jl

```diff
@@ -594,3 +595,22 @@ relpath(path::AbstractString, startpath::AbstractString) =
 for f in (:isdirpath, :splitdir, :splitdrive, :splitext, :normpath, :abspath)
     @eval $f(path::AbstractString) = $f(String(path))
 end
+
+"""
+    username() -> String
+
+    Return the “login name” of the user.
+    This function checks the environment variables LOGNAME, USER, LNAME and USERNAME, in order, and
+    returns the value of the first one which is set to a non-empty string.
+    If none are set, the login name from the password database is returned
+"""
+
+function username()
+    for varname in ("LOGNAME", "USER", "LNAME",  "USERNAME")
+        username = get(ENV, varname, "")
+        !isempty(username) && return username
+    end
+
+    uv_passwd = Libc.getpwuid(Libc.getuid(), true)
+    return uv_passwd.username
```

Comment:
```
julia> getusername
ERROR: UndefVarError: getusername not defined
```

Reply:
Just do `using Base.Filesystem: getusername` to test this at the REPL.

Reply:
`Base.Filesystem.getusername` is not type-stable (it can return `nothing` if the `username` field is empty).   I suspect that we may want the lower-level `getpwuid` here, and maybe throw an error if the username string is empty.